app/commands.py

```diff
@@ -1321,6 +1321,34 @@ async def debug(ctx: Context) -> Optional[str]:
     return f"Toggled {'on' if app.settings.DEBUG else 'off'}."
 
 
+@command(Privileges.ADMINISTRATOR, hidden=True)
+async def givedonator(ctx: Context) -> Optional[str]:
+    """Gives donator to a specified player (by name) for a specified time, such as '3h5m'."""
+    if len(ctx.args) < 2:
+        return "Invalid syntax: !setdonator <name> <duration>"
+
+    if not (t := await app.state.sessions.players.from_cache_or_sql(name=ctx.args[0])):
+        return "Could not find user."
+
+    seconds = timeparse(ctx.args[1])
+
+    if seconds is None:
+        return "Invalid timespan."
+
+    if seconds < 5 * 60:
```

Comment:
I implemented it so that people are not confused when donator isn't gone after 5 seconds because of the 5min background task

Reply:
it will last the duration it is supposed to lol, if they expect donor to instantly expire when they specified `5 minutes` that is a problem of their own